### DISCUSSION TITLE
Some features work, some don't

### DISCUSSION THREAD

**Discussion by Magna-Graecia** [rodit/SnapMod#96] [General]
Hey mate. Thanks so much for working on this. This has been sorely needed in the snapchat modding community.

So the app is awesome and well-made, but unfortunately only some of the features work for me, although it's quite feature-packed!

I'll start by listing my device and software info, as well as the features that WORK and DON'T WORK. Then I'll state what solution I've tried.

My device is an SM-T580 Samsung Galaxy Tab A (2016) running Android 8.1.0 and rooted with Magisk 23.0. The Magisk app is also 23.0. I didn't have any interest in upgrading beyond this version. I'm using Systemless Xposed v90.2-beta3 with Xposed Installer v3.1.5. I know your readme says that you have no interest in supporting other Xposed implementations, but SnapMod should work fine with them either way, and it does for the most part in my case. However, I thought the fact that this is specifically developed around the LSPosed framework might be why it wasn't fully working properly, so I did try going the route of using LSPosed instead, which I will describe after the list of working/not working features.

List of working/not working features:

**- Disable Camera:** WORKS
**- Hide Screenshot/Record Notifications:** WORKS
**- Hide Saved to Camera Roll:** WORKS
**- Hide Replay Notifications:** DOES NOT WORK
**- Hide Typing Status (Bitmoji):** DOES NOT WORK
**- Hide Friends:** WORKS
**- Anonymous Story Viewing:** Previously did not work, but after the latest update today this now works with both this and Ignore Story Viewing (Local) enabled.
**Ignore Story Viewing (Local):** WORKS
**- Disable Some Metrics and Logging:** I haven't tried this, so I can't comment on it.
**- Hide Read Receipts:** This works for chats. Does not work for snaps. Not sure if it's supposed to, but one workaround for this is to enable both "Hide Saved to Camera Roll" and "Save Any Message", and then you can open a chat by pressing the chat bubble icon on the right side of any chat in your chat list (so you don't open the snap), and then pressing and holding on the snap within the chat to get the context menu to save to camera roll. This allows you to both save and see the snap without ever opening it, therefore no read receipt haha.
**- Hide Chat Saves/Unsaves:** DOES NOT WORK. In fact, enabling this causes me to no longer be able to save/unsave any chats by tapping on them with my finger. If I press and hold on them to get the context menu and select "save/unsave", it does nothing. I have to disable this to get the saving/unsaving feature back.
**Hide Enter Conversation (Bitmoji)/Hide Exit Conversation (Bitmoji):** Neither of these work, which is very unfortunate because this not working sort of ruins hide read receipts in a way. If you end up opening a message with hide read receipts on while the person is still in the chat, then sure the "Delivered" icon will show in their chat list, signaling that you haven't read it, but they literally just saw you pop up in chat, and even exit perhaps lol. Thereby ruining any "stealth" aspect. Also, if I have these both enabled and I go into one of my chat windows, all messages will be gone. It will just be blank. If I disable them and go back to the chat window, all messages will now be there again. Strange.
**Don't Release Messages:** DOES NOT WORK
**Save Any Message:** WORKS. One thing I'd like to note is that you should state in the readme that people should go and make sure storage permissions are granted to SnapMod within settings in order for any saving to work because mine wasn't, and I thought saving wasn't working until I manually enabled it. Might save some unnecessary new issues being opened.
**Allow Pinning Chats:** WORKS
**Block Ads:** I haven't tried this, so I can't comment on it.
**Disable Spotlight Tab:** I haven't tried this, so I can't comment on it.
**Hide Story Sections:** I haven't tried this, so I can't comment on it.
**Disable Bitmojis:** WORKS
**Override Add Friend Method:** I haven't tried this, so I can't comment on it.
**Send Gallery as Snaps:** WORKS. However, sending to story will cause "from camera roll" to show.
**Allow Saving Snaps:** WORKS
**Override Gallery Snap Type:** DOES NOT WORK. If I enable this while having "Send Gallery as Snaps" enabled, all snaps will be sent in chat instead of live. It causes "Send Gallery as Snaps" to be disabled basically.
**Bypass Video Length Restrictions:** WORKS. However, I should note that it works whether this is enabled or not. Not sure why. Is it not supposed to?? I tested sending the same 47 second video from another device that doesn't have SnapMod and isn't even rooted or modded in any way, shape, or form, and it sent through chat just fine. Not live like it will with SnapMod, but ya. What is the limit? I was under the assumption that it was 11 seconds. I've definitely sent a 47 second video both with this enabled and disabled, and it sent live just fine. In fact, I don't know if I'm just seeing things, but I felt like the quality was better when it was disabled. I may just be imagining things on that one though haha. If I try to share a video from gallery to snapchat that's longer than 11 seconds though, I get the restriction error saying it's longer than 11 seconds, and it won't load into the camera window. I can only do it from the gallery drawer in chat, which sucks because you have to scroll through there since there's no search.
**Unlimited Snap/Story Duration:** WORKS
**Allow Download Stories:** WORKS
**Allow Download Public DPs:** Not sure if this one works. I did try to go to Kourtney Kardashian's public profile just to test, and I held down on their profile picture, but nothing happened. No photos were downloaded. I checked storage. Maybe this only works on public profiles that have added you back as a friend or something?
**Use Android Download Manager:** I just wanted to note that I do have this enabled. I'm having no issues with saving, so figured I'd leave it enabled.
**Location Sharing Override:** I haven't tried this, so I can't comment on it. Personally I would say not to focus any efforts on supporting this feature. It's pointless on Android as there are so many apps that will do this in an even easier fashion, and with more options like being able to fake movement and follow specific routes. You can literally make your bitmoji fly from L.A. to London in an airplane bitmoji by creating a route and changing speed and altitude.
**Show More Info on Profile:** WORKS
**Log Network Requests:** Haven't had a need to try this, so can't comment on it.

So that's it. Now, as for the solution I tried. I basically disabled Systemless Xposed from within Magisk, then installed both Riru v26.1.4 and LSPosed v1.8.2. Installation of both went fine, and LSPosed said "v1.8.2 active" or whatever at the top of the manager. No error messages. I also made sure SnapMod was active within it, and that Snapchat was selected within its subcategory, and I also did plenty of reboots between doing things to refresh things. None of the SnapMod features worked. One thing I noticed was the camera was disabled as if I had the "Disable Camera" option enabled even though I didn't. After disabling both Riru and LSPosed and re-enabling Systemless Xposed (and rebooting where necessary of course), my camera came back and everything worked as before (some features working, some not, as stated above).

Anyway, I hope this helps enlighten you as to how it's working on various devices/software. If you need anything else from me, let me know. And if you have any idea on how to fix my issues, please let me know as well. Thanks again for working on this, and sorry the post is so long. I just wanted to be as thorough as possible :)

**Comment by rodit** [comment]
Thank you for taking the time to test everything - I appreciate the detailed report. I am aware that most of the experimental features don't work (the test tube labelled ones in the app, especially modifying read receipts/view states) and will eventually get round to fixing them (or removing them entirely if they are too difficult to get working).

Just a few notes: The 'Allow Download Public DPs' currently only works for friends profile pictures (when you hold down the thumbnail for their profile picture on their friend card rather than on the actual public profile screen), so what you experienced is expected.

In regards to the video length bypass feature, from my understanding, the quality is lowered for snaps if they exceed a certain length - maybe this isn't in place for gallery media.

I'm not quite sure why it wouldn't work with LSPosed but there are no LSPosed specific features in SnapMod so it should, as you have shown, work with original Xposed. SnapMod automatically disables the camera on start (as it can't load the config before Snapchat opens the camera) and only re-enables the camera if the config says it should when it loads which is why the camera was disabled for you. There is probably some logcat or LSPosed log output that would shed some light on why it wasn't working.

**Comment by snaphappy1** [comment]
I have similar setup, Android 8.1 systemless original exposed.

**- Disable Camera:** WORKS
**- Hide Screenshot/Record Notifications:** WORKS
**- Hide Saved to Camera Roll:** WORKS
**- Hide Replay Notifications:** BROKEN, not too important
**- Hide Typing Status (Bitmoji):** BROKEN kinda, bitmoji in chat still shows I'm typing, but this option does prevent other person from receiving '...is typing' notifications
**- Hide Friends:** WORKS
**- Anonymous Story Viewing:** WORKS
**- Ignore Story Viewing (Local):** WORKS
**- Disable Some Metrics and Logging:** haven't tried...
**- Hide Read Receipts:** WORKS
**- Hide Chat Saves/Unsaves:** BROKEN, not too important 
**- Hide Enter Conversation (Bitmoji):** BROKEN
**- Hide Exit Conversation (Bitmoji):** BROKEN
**- Don't Release Messages:** BROKEN, not too important
**- Save Any Message:** WORKS
**- Allow Pinning Chats:** WORKS
**- Block Ads:** WORKS
**- Disable Spotlight Tab:** WORKS
**- Hide Story Sections:** WORKS
**- Disable Bitmojis:** WORKS
**- Override Add Friend Method:** Didn't work with an earlier version, got temp banned. Haven't tried since lol
**- Send Gallery as Snaps:** WORKS 
**- Allow Saving Snaps:** WORKS
**- Override Gallery Snap Type:** Haven't tried...
**- Bypass Video Length Restrictions:** Haven't tried...
**- Unlimited Snap/Story Duration:** WORKS
**- Allow Download Stories:** WORKS
**- Allow Download Public DPs:** Haven't tried...
**- Use Android Download Manager:** WORKS. Haven't tried disabling...
**- Location Sharing Override:** Haven't tried...
**- Show More Info on Profile:** WORKS
**- Log Network Requests:** Haven't tried.

Overall most important features work.
I tried using snapmod on a different phone running Android 9 pie and Lsposed/Riru, didn't work.

**Comment by Magna-Graecia** [comment]
> Thank you for taking the time to test everything - I appreciate the detailed report. I am aware that most of the experimental features don't work (the test tube labelled ones in the app, especially modifying read receipts/view states) and will eventually get round to fixing them (or removing them entirely if they are too difficult to get working).
> 
> Just a few notes: The 'Allow Download Public DPs' currently only works for friends profile pictures (when you hold down the thumbnail for their profile picture on their friend card rather than on the actual public profile screen), so what you experienced is expected.
> 
> In regards to the video length bypass feature, from my understanding, the quality is lowered for snaps if they exceed a certain length - maybe this isn't in place for gallery media.
> 
> I'm not quite sure why it wouldn't work with LSPosed but there are no LSPosed specific features in SnapMod so it should, as you have shown, work with original Xposed. SnapMod automatically disables the camera on start (as it can't load the config before Snapchat opens the camera) and only re-enables the camera if the config says it should when it loads which is why the camera was disabled for you. There is probably some logcat or LSPosed log output that would shed some light on why it wasn't working.

You're welcome! And thank you for the informative reply. Oh ok, so this is as expected. Well it's good to know that it isn't just a "me" problem then haha. You just launched this, so we can't expect everything to be perfect right from the get-go. It's already working pretty damn good as it is, so hopefully in due time you can iron out the kinks. As snaphappy1 said, most important features work, so that's the most important thing. In all honesty, nearly everything I'm interested in is already working. The only big one that I would really like working is "Hide Enter Conversation/Hide Exit Conversation" because it works well in combination with "Hide Read Receipts". SnapFreedom actually combined these 2 features into one option called "chat stealth". If you enabled it, then you could read messages without them ever saying they were opened, and your bitmoji also never appeared for the other person, whether entering or exiting. If you tried typing while this option was enabled, then snapchat crashed lol, which is a good thing to be honest because you wouldn't want to be typing and replying to a message that shows as never even having been opened yet for the other person. 

Other features have easy workarounds to them. Take the restriction for sharing videos longer than 11 seconds from gallery to snapchat so you can send them as live snaps for example. Sharing from gallery to snapchat is much more convenient because going through the in-chat gallery drawer means you need to scroll through countless photos/videos just to find the specific one you want, and there's no filter or search option. This already works fine for photos and videos shorter than 11 seconds, and they get sent as live snaps, but unfortunately when you try to share a video longer than 11 seconds you get an error message saying the video exceeds the time limit. So the only option for accomplishing this is to use the in-chat gallery drawer, which has the issue explained above. So how do you get around this? Find the video in your gallery, then make a copy of it in the same folder, or even a different one. Once you do this, the current date and time is recorded for it, which means that it now pops up as the first item in the in-chat gallery drawer since it's the most recent one, so no more scrolling for minutes on end. You have to of course go back and make sure you delete the copies after you're finished, unless you don't care about your storage filling up with useless duplicates haha, so it's not the most convenient method, but it's easy enough and works fine.

Another workaround example would be "Hide Replay Notifications". You don't really need this if you use "Hide Saved to Camera Roll" in combination with "Save Any Message" or "Allow Saving Snaps". If you view a snap and accidentally close it and now want to replay it but don't want the person receiving a notification, then no problem. Just press and hold the "opened snap" bubble in the chat to get the context menu, then select "save to camera roll". You now have the snap saved on your device and can do what you please with it, and the person never received a notification, and you didn't need to replay it. The same logic applies to another feature that's missing in SnapMod, and that's stealth snap viewing. SnapFreedom and other snapchat modding tools had a feature like this where you can open a snap, then close it, and it would appear like it was never opened. It's basically a "hide read receipt" for snaps. You can do the same method explained above to save/view snaps before even opening them, so boom, you now have stealth view.

There's a workaround to another issue I mentioned in my original post, and that someone else also mentioned in an issue they opened. In fact, I'm going to reply to their post after this to let them know. This issue is the one where sending gallery snaps to "my story" means they have a "from camera roll" tag on them instead of showing as live. The solution to this is another easy one. Just open a chat, open the gallery drawer, and then press and hold on the photo/video you want to share. This will cause it to load into the camera window. From there, hit the "save/download" button in the bottom left-hand corner. This will add the photo/video to the snaps section of memories with a recording of the current date, time, and location. Close out of that snap, open the one in memories, and send it to your story, It now won't show "from camera roll".

Thanks for letting me know about the "Allow Download Public DPs"! This is understood.

Ya I think I was just imagining a quality difference between having "Bypass Video Length Restrictions" enabled or disabled. I was seeing things haha. From my understanding though, I don't think it's the length of the video that affects the quality, but the size of it. I have a lot of experience using these snapchat modding tools since back in the early days of snap. I used to use something back then, and I forgot its name, but it wasn't very feature-packed. It was just for sharing photos/videos as live snaps, as well as stealth saving. Then the first feature-rich one I started using was Snapprefs years ago, which was I think the first Xposed module based one. That was abandoned, and SnapFreedom took over. SnapFreedom was abandoned, and now you're taking over. Don't leave us, bro! Lmao :P .. But anyway, so in my experience of having used all of these over the years, basically the larger the video size is, the more snapchat compresses the crap out of it and makes it look like garbage. I would notice huge differences if I took a video and reduced its size on my PC using various programs like windows movie maker or even handbrake that accomplished this without lowering the quality. Snapchat would compress it so much less. There's also definitely a difference in quality between sending a video as a live snap and posting it to your story. Story gets compressed so much more, and this makes sense since it's being hosted on Snapchat's servers for 24 hours, so they're going to try to reduce its size as much as possible. Snaps on the other hand I believe get temporarily downloaded on the receiver's device and are not being hosted in the cloud or anything. This is why you can do the trick where you receive a snap, enable airplane mode (connection doesn't matter since the snap is already downloaded), open the snap, then clear Snapchat's data in settings before disabling airplane mode and re-opening snapchat. You of course need to log in again, but the snap you opened, viewed, screenshot, whatever will show as though you never opened it. The same goes for chats, viewing/screenshotting stories, or really anything you do on snap. Using a mod tool like SnapMod is so much better though of course haha. That method is a pain in the ass and not really practical.

Ya the camera is disabled when I first open snapchat, which prompts snapchat to give the message saying "snapchat is a camera app and uses the camera blah blah blah". I just press cancel, and of course the camera loads fine unless I have "disable camera" enabled in SnapMod's settings. However, when I tried using LSPosed, the camera would be disabled when opening snapchat and would remain disabled, even though I didn't have the option enabled in SnapMod's settings. Every time I would scroll between snapchat's different sections (snapmap, chats, camera, etc), and land on the camera page, it would remain black and I would get that message once again. Oh well, not a biggie since I'll be using original Xposed anyway, which works as intended. As for why it wasn't really working with LSPosed, I'm not sure why, but I found it interesting that snaphappy1 pointed out in his reply that he tried using SnapMod on a different device that was running Android 9 pie and Riru/LSPosed, and it didn't work. His experiences have mirrored mine completely. SnapMod also didn't work for me with Riru/LSPosed, although I tried it on Android 8.1.0 and not 9. And the features that are working/not working for him while running on Android 8.1.0 with original systemless Xposed (same as me) are the EXACT same ones that are working/not working for me. I even went and tested what he wrote about "Hide Typing Status", and it works exactly as he said. When enabled, it doesn't hide the bitmoji showing as typing in snapchat, but it does prevent the receiver from receiving any "... is typing" notifications on the home screen of their device, or if their device screen is locked. I tested both ways enabling and disabling "Hide Typing Status". With it enabled, the device I was typing a message to never received any notifications on the home screen or with the screen locked, but I could see the bitmoji typing if I opened snapchat. With it disabled, the device I was typing a message to did receive a notification complete with audio on the home screen. It also played the audio sound and woke the screen when the device was locked. Very interesting!

Anyway, once again sorry for the long posts. Just trying to inform you as thoroughly as possible. Thanks for your work, and I'm looking forward to future updates!

**Comment by Magna-Graecia** [comment]
> I have similar setup, Android 8.1 systemless original exposed.
> 
> **- Disable Camera:** WORKS **- Hide Screenshot/Record Notifications:** WORKS **- Hide Saved to Camera Roll:** WORKS **- Hide Replay Notifications:** BROKEN, not too important **- Hide Typing Status (Bitmoji):** BROKEN kinda, bitmoji in chat still shows I'm typing, but this option does prevent other person from receiving '...is typing' notifications **- Hide Friends:** WORKS **- Anonymous Story Viewing:** WORKS **- Ignore Story Viewing (Local):** WORKS **- Disable Some Metrics and Logging:** haven't tried... **- Hide Read Receipts:** WORKS **- Hide Chat Saves/Unsaves:** BROKEN, not too important **- Hide Enter Conversation (Bitmoji):** BROKEN **- Hide Exit Conversation (Bitmoji):** BROKEN **- Don't Release Messages:** BROKEN, not too important **- Save Any Message:** WORKS **- Allow Pinning Chats:** WORKS **- Block Ads:** WORKS **- Disable Spotlight Tab:** WORKS **- Hide Story Sections:** WORKS **- Disable Bitmojis:** WORKS **- Override Add Friend Method:** Didn't work with an earlier version, got temp banned. Haven't tried since lol **- Send Gallery as Snaps:** WORKS **- Allow Saving Snaps:** WORKS **- Override Gallery Snap Type:** Haven't tried... **- Bypass Video Length Restrictions:** Haven't tried... **- Unlimited Snap/Story Duration:** WORKS **- Allow Download Stories:** WORKS **- Allow Download Public DPs:** Haven't tried... **- Use Android Download Manager:** WORKS. Haven't tried disabling... **- Location Sharing Override:** Haven't tried... **- Show More Info on Profile:** WORKS **- Log Network Requests:** Haven't tried.
> 
> Overall most important features work. I tried using snapmod on a different phone running Android 9 pie and Lsposed/Riru, didn't work.

Thanks so much for this input! This was very informative. I tested what you wrote about "Hide Typing Status" partially working, and you're right, it works exactly as you stated. I had never noticed that detail before. It's very interesting that your experiences mirror mine completely. When using the same OS and Xposed version, the features that work and don't work for you are exactly the same as me. Then when you tried using SnapMod on a device running Riru/LSPosed, it didn't work for you, which was exactly my experience when trying to use it with Riru/LSPosed. The only difference is I tried that on 8.1.0, not Android 9. I wonder if others have everything working, including "Hide Typing Status", "Hide Enter Conversation/Hide Exit Conversation", etc. By the way, go through my reply above to rodit for workaround tips in case you weren't aware of these. I explained some workarounds there for "Hide Replay Notifications", how to stealth view a snap, and how to have a snap sent to story not show "from camera roll". It might help if you weren't already aware.

I found the "Override Add Friend Method" super interesting, but hadn't tried it yet. But if you got temp banned from using it, then I probably won't lol. It could be pretty funny to make people think they added you and you added them back instead of the other way around though haha. Like "huh? When tf did I add this person? Who is this?!?" XD

**Comment by Magna-Graecia** [comment]
@rodit Just wanted to update you on a couple of interesting things I found out in testing.

So in my reply to you above I had mentioned a workaround to getting snaps sent to story to not show "from camera roll", and I even replied to thracianx with this workaround because he had posted this issue. Welp, I found out that you actually CAN send straight from gallery as a "live" snap to story. So here is how it works. If you SHARE a photo/video from gallery to snapchat so that it loads into the camera window, then send to story, it WILL show "from camera roll". However, if you bring up the gallery drawer in a chat and press and hold on a photo/video so that it loads into the camera window, then send to story, it does NOT show "from camera roll." No need to load a photo/video into the camera window and press the save/download button in the bottom left to save it to memories, then send from there.

This brought up another interesting tidbit when trying to use this with videos longer than 11 seconds. So in my original post I had mentioned how "Bypass Video Length Restrictions" wasn't necessary for me because whether it was enabled or not, I was still able to send videos longer than 11 seconds as live snaps directly from gallery drawer. If I brought up the gallery drawer in a chat, then selected a video by tapping on it to get the checkmark, and hit send, then this is true. HOWEVER, if I pressed and HELD on a video longer than 11 seconds to get it to load into the camera window, I would get an error message saying "Videos longer than 10 seconds can't be edited." Why snapchat sometimes says 10 seconds and other times says 11 seconds, I have no idea, but the point is that this prompted me to go back to SnapMod settings and enable "Bypass Video Length Restrictions." After enabling this, killing snap, and restarting it, I was now able to press and hold on a video longer than 10/11 seconds in gallery drawer and get it to load into the camera window. Which means I cannot only edit it, but I can send it to story without any "from camera roll" tags.

So "Bypass Video Length Restrictions" DOES work. It's not necessary for tapping on a video in gallery drawer and then directly sending (it will get sent as a live snap whether it's enabled or not), but if you DO want to press and HOLD on the video to load it into the camera window for edits and sending to story, then it does need to be enabled, and it's working beautifully.

Also, one last thing. We had also discussed video quality when sending snaps. So as you know, when you record a video longer than 10 seconds, snapchat chops the video up into multiple 10 second snaps and sends this way. This obviously isn't all that ideal because you see cuts in the full snap, and you especially hear it in the audio. So what I noticed when using SnapMod is that if I bring up the gallery drawer in a chat, select a video longer than 10 seconds by tapping on it, then hitting the send button, the video does NOT get chopped up. It gets sent as one full live snap. I tested this with the same 47 second video mentioned previously. It got sent as one 47 second-long snap. The benefit of this is that snapchat doesn't touch the video with any edits, so the quality ends up being higher. This is what I noticed in testing at least. There's also of course no visual or audible cuts. Pretty neat! However, if you bring up gallery drawer in a chat, press and HOLD on a video longer than 10 seconds to get it to load into the camera window (with Bypass Video Length Restrictions enabled of course), THEN hit send, the video WILL get chopped up into multiple 10-second long segments. I found this interesting, and it's pretty neat because it gives you options on whether you want your snap to be one full-long snap, or cut into 10-second long segments like it normally is. Obviously if you want to add any edits to the video, or send it to story, then you will need to load it into the camera window, which means it will get chopped up regardless. But it's nice to know we at least have options.

Hope this info helps give some insight.

**Comment by rodit** [comment]
Thank you again for the info - I was aware of the sending to story not showing the 'From Camera Roll' text but I wasn't sure if this was just how Snapchat was supposed to work or not.

I will probably officially document these findings at some point so people know how to use the features properly.

That's also good to know - with the new(ish) multisnap feature on Snapchat, you can combine multiple snaps into one longer snap anyway, which is probably why you can send >10 second unbroken snaps from your gallery.

**Comment by Magna-Graecia** [comment]
> Thank you again for the info - I was aware of the sending to story not showing the 'From Camera Roll' text but I wasn't sure if this was just how Snapchat was supposed to work or not.
> 
> I will probably officially document these findings at some point so people know how to use the features properly.
> 
> That's also good to know - with the new(ish) multisnap feature on Snapchat, you can combine multiple snaps into one longer snap anyway, which is probably why you can send >10 second unbroken snaps from your gallery.

You're welcome. Evidently this is just how Snapchat operates. I tested on a device that isn't rooted or modded in any way at all, and when sharing from gallery to snapchat, then sending to story, I got "from camera roll", but if I went into a chat, opened the gallery drawer, pressed and held on a photo/video to load it into the camera window, then sent, it wouldn't show "from camera roll." Pretty funny, but hey it works.

So I just tested multisnap out (never actually used it before), and it still has the breaks. It doesn't actually morph multiple snaps into one seamless video like we can do right now with SnapMod. All it basically does is allow you to take one snap, go do other stuff, come back and take another snap, go do some more other stuff, come back and take another snap, etc, and when you're ready to send, they'll all get sent at once. They'll still be broken up exactly as you took them. So for example I just tested by recording 2 video snaps approximately 15 seconds long each. Well not only did it not combine them into one seamless 30 second long video, but it didn't even combine them into 3 10-second long snaps. A 15 second long snap is 2 snaps, so recording 2 of those will be 4 snaps. Multisnap kept it at 4 snaps lol. It's a pretty useless feature, which is why I never used it. It seems like its use is so that you can take multiple snaps at different moments, and do things in between, while not having to immediately send after taking each one because snapchat holds onto them in memory when the multisnaps feature is enabled. You can then send them all at once when you're ready. Pretty useless for me at least.

As of right now, I'm not aware of any way to truly send a minute-long video as one single seamless snap with no breaks whatsoever without the use of SnapMod. There was a way to do this with SnapFreedom as well, but it wasn't intended. It was a glitch, but once I found out what caused the glitch, I was able to reproduce it whenever I wanted if desired. Right now on SnapMod we have the option to send either way as explained in my post above, so I think that's pretty cool haha

**Comment by NMBlanco** [comment]
@Magna-Graecia thank you for the insights, very helpful!

**Comment by Magna-Graecia** [comment]
> @Magna-Graecia thank you for the insights, very helpful!

You're welcome! Glad you were able to gain something from it all :)

**Comment by NMBlanco** [comment]
@Magna-Graecia what I noticed with trying to save snaps immediately AFTER you viewed them:
Sometimes, it gives you the option to long press the viewed snap in chat and then there will be the drop down menu where you can either choose to replay it, delete it, or save it to camera roll. It won't automatically replay the snap though, which is good. 

But very often I somehow only get the classic option to replay the snap after I viewed it. There won't be a menu from where I can choose what to do with it. When I long press the viewed snap, it just replays and sends the notification out, even though its disabled.
Have you also noticed this?

**Comment by Magna-Graecia** [comment]
@tarek133 Hmm, strange. I haven't noticed this myself. "Hide Replay Notifications" is broken for me too. It still sends notifications even when the option is enabled in SnapMod settings, but I haven't run into the problem you described where I don't get the context menu on an opened snap when long pressing. I would definitely recommend enabling "Unlimited Snap/Story Duration" so you can view any snap for as long as you like and be able to decide if you want to save it right then and there using the context menu when pressing the 3 dots in the upper right-hand corner. This way you don't need to save afterward by long pressing and running the risk that it replays and sends a notification. I would do this as a temporary stop gap. In the meantime, open an issue with the problem you described. It seems like a legitimate issue that should be brought to rodit's attention. Make sure to give him your device info, Android version, Xposed version (original Xposed, LSPosed, etc), SnapMod version, Snapchat version, and an Xposed/LSPosed log. He may be able to figure out the problem and address it.

If you don't want to do that immediately, something I would definitely do is monitor it yourself by immediately checking the Xposed/LSPosed log the next time it replays instead of giving you the context menu when you long press. It might show an error in the log. This is what rodit would be looking for I assume.
